airbyte-integrations/base-java/src/main/java/io/airbyte/integrations/javabase/Destination.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.javabase;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.config.DestinationConnectionSpecification;
+import io.airbyte.config.Schema;
+import io.airbyte.config.StandardCheckConnectionOutput;
+import io.airbyte.config.StandardDiscoverSchemaOutput;
+import io.airbyte.singer.SingerMessage;
+import java.io.IOException;
+
+// todo (cgardens) - share common parts of this interface with source.
+public interface Destination {
```

Comment:
we don't actually have a type for CONFIG though. right now it's not a class that we generate from the jsonschema spec (though we could).

Reply:
I'm coming from the perspective that we would want to manually define or generate a class for configs in most cases. It feels more natural to extract the data from the config json into an object than to manually access all the properties.

However, if someone really doesn't want to deal with this, we would be able to still allow `JsonNode` as the config type, and it would work identically to the current version.

Reply:
fair. i'll add an abstract class to do this. going to add it as a todo for now.